pkg/sanity/node.go

```diff
@@ -345,15 +278,53 @@ var _ = Describe("NodeUnpublishVolume [Node Server]", func() {
 					Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
 				},
 			},
+			StagingTargetPath: csiStagingTargetPath,
 		}
 		if controllerPublishSupported {
-			nodepubvolRequest.PublishInfo = conpubvol.GetPublishInfo()
+			nodeStageVolReq.PublishInfo = conpubvol.GetPublishInfo()
 		}
-		nodepubvol, err := c.NodePublishVolume(context.Background(), nodepubvolRequest)
+		nodestagevol, err := c.NodeStageVolume(
+			context.Background(), nodeStageVolReq)
 		Expect(err).NotTo(HaveOccurred())
-		Expect(nodepubvol).NotTo(BeNil())
+		Expect(nodestagevol).NotTo(BeNil())
+	}
+	// NodePublishVolume
+	By("publishing the volume on a node")
+	nodepubvolRequest := &csi.NodePublishVolumeRequest{
+		VolumeId:   vol.GetVolume().GetId(),
+		TargetPath: csiTargetPath,
+		VolumeCapability: &csi.VolumeCapability{
+			AccessType: &csi.VolumeCapability_Mount{
+				Mount: &csi.VolumeCapability_MountVolume{},
+			},
+			AccessMode: &csi.VolumeCapability_AccessMode{
+				Mode: csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER,
+			},
+		},
+	}
+	if nodeStageSupported {
+		nodepubvolRequest.StagingTargetPath = csiStagingTargetPath
+	}
+	if controllerPublishSupported {
+		nodepubvolRequest.PublishInfo = conpubvol.GetPublishInfo()
+	}
+	nodepubvol, err := c.NodePublishVolume(context.Background(), nodepubvolRequest)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(nodepubvol).NotTo(BeNil())
+
+	// NodeUnpublishVolume
+	By("cleaning up calling nodeunpublish")
+	nodeunpubvol, err := c.NodeUnpublishVolume(
+		context.Background(),
+		&csi.NodeUnpublishVolumeRequest{
+			VolumeId:   vol.GetVolume().GetId(),
+			TargetPath: csiTargetPath,
+		})
+	Expect(err).NotTo(HaveOccurred())
+	Expect(nodeunpubvol).NotTo(BeNil())
 
-		// NodeUnpublishVolume
+	if controllerPublishSupported {
```

Comment:
A bit confusing here, the check if for controllerPublishSupported, but the call is for NodeUnpublishVolume.

Reply:
good catch. This was just copy pasted from the old code.

I have also added an Unstage step as well if nodeStageSupported.